app/src/main/java/com/mstudio/superheroarch/MainActivity.kt

```diff
@@ -40,19 +43,26 @@ class MainActivity : AppCompatActivity() {
         call.enqueue(object : Callback<CharactersResponse> {
             override fun onResponse(call: Call<CharactersResponse>, response: Response<CharactersResponse>) {
                 if (response.isSuccessful) {
-                    if (response.body() != null) {
-                        titleToChange = findViewById(R.id.title_to_change)
-                        titleToChange.text = response.body()?.toString()
-                        Log.d("Successful response", response.body().toString())
-                    }
-                    Log.d("Successful response", response.body().toString())
+                    val firstCharacter = response.body()?.results?.firstOrNull()
+                    firstCharacter?.let {
+                        characterNameTitle = findViewById(R.id.character_name)
```

Comment:
if you do here the findViewById you'll do it every time the button is pressed. 

You are already declaring the TextViews outside. Just assign them the values at the onCreate to avoid the needed of search the reference every time the button is pressed.